docs/cody/proxy-setup.mdx

```diff
@@ -0,0 +1,112 @@
+
+```json
+{
+  "http.proxy": "http://proxy_ip:port", // Replace with your proxy IP and port
+  "http.proxySupport": "on",
+  "http.proxyStrictSSL": true,
+  "http.systemCertificates": true,
+  "http.experimental.systemCertificatesV2": true
+}
+```
+
+### Self-Signed Certificates in VS Code
+If your proxy uses self-signed certificates, ensure the following settings are enabled:
+
+```json
+{
+  "http.systemCertificates": true,
+  "http.experimental.systemCertificatesV2": true
+}
+```
+
+### Restart VS Code
+After modifying the `settings.json` file, completely close and restart VS Code to apply the new proxy settings.
+
+### Supported Proxy Types
+VS Code supports HTTP, HTTPS, and SOCKS proxies. The `http.proxy` setting in `settings.json` will accommodate all three types based on the provided proxy URL.
+
+## Proxy Setup for JetBrains
+
+### Accessing the Proxy Settings
+1. Open your JetBrains IDE (e.g., IntelliJ IDEA, PyCharm).
+2. Navigate to `Settings` by clicking on `File` > `Settings` (or `Preferences` on macOS).
+3. In the search bar at the top, type `proxy` to find the proxy settings.
+4. Select the `HTTP` proxy or `SOCKS` proxy based on your requirement.
+
+### Configuring HTTP Proxy
+1. In the proxy settings, choose `Manual Proxy Configuration`.
+2. Enter the following details:
+    - **HTTP Proxy:** `http://proxy_ip:port` (Replace with your proxy IP and port)
+    - **Enable proxy authentication if needed and provide credentials.**
+3. Click `Apply` and then `OK` to save the settings.
+
+### Configuring SOCKS Proxy
+1. In the proxy settings, choose `Manual Proxy Configuration`.
+2. Enter the following details:
+    - **SOCKS Proxy:** `socks5://proxy_ip:port` (Replace with your proxy IP and port)
+    - **Enable proxy authentication if needed and provide credentials.**
+3. Click `Apply` and then `OK` to save the settings.
+
+### Using the Proxy with Cody
+1. Open Cody settings within your JetBrains IDE.
+2. Connect to the Sourcegraph instance using the provided instance IP and token:
+    - **Instance IP:** `http://instance_ip`
+    - **Token:** `your_token`
+3. Click `Apply`, then `OK`, and test the connection.
+
+### Supported Proxy Types
+JetBrains supports HTTP and SOCKS proxies. The `Manual Proxy Configuration` in the settings will accommodate both types based on the provided proxy URL.
+
+**Note:**
+1. Only `Manual Proxy Configuration` is supported at the moment. The `auto-detect proxy settings` option is not fully supported.
+2. HTTPS proxy is not supported in JetBrains currently.
+
+## Troubleshooting the Proxy Setup
+
+### Basic Connectivity Test
+To verify that your proxy setup works, use the following `curl` command in your terminal, replacing `proxy_ip` and `instance_ip` with your actual proxy and instance IP addresses:
+
+```bash
+curl -x http://proxy_ip:port http://instance_ip
+```
+
+This command should return a response from the instance, confirming that the proxy is correctly routing your requests.
```

Comment:
@arafatkatze Thank you for doing this! This makes troubleshooting super easy! Nicely done!

cc: @RXminuS when we think about Cody Doctor, we should have some check for Proxy setup too :)